ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/HeartbeatAgent.java

```diff
@@ -157,37 +154,18 @@ void stopAgent() {
 
   void setLocalAddress(final String applicationServer) {
 
-    this.localHostInfo = parseHostInfo(applicationServer);
-    this.localHostString = localHostInfo.toString();
+    this.localHostInfo = ServerUtil.parseHostInfo(applicationServer);
     try {
       this.localURL = new URL(applicationServer);
     } catch (final Exception e) {
       throw new IllegalStateException("Failed to convert remote host info to URL."
                                           + " remoteInfo: " + localHostInfo.host() + ":"
                                           + localHostInfo.host());
     }
-    this.hostsStatus.putIfAbsent(localHostString, new HostStatusEntity(
-        new HostInfoEntity(localHostInfo.host(), localHostInfo.port()),
-        true,
-        clock.millis()));
+    //This is called on startup of the heartbeat agent, no other entries should exist in the map
```

Comment:
nit: instead of writing this in a comment, enforce it 😄 
```java
Preconditions.ensureState(immutableHostsStatus.get().isEmpty(), "expected empty host info on startup");
```